src/amo/components/SearchFilters/index.js

```diff
@@ -0,0 +1,185 @@
+import { oneLine } from 'common-tags';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+
+import {
+  ADDON_TYPE_EXTENSION,
+  ADDON_TYPE_THEME,
+  OS_LINUX,
+  OS_MAC,
+  OS_WINDOWS,
+} from 'core/constants';
+import { withErrorHandler } from 'core/errorHandler';
+import log from 'core/logger';
+import translate from 'core/i18n/translate';
+import { convertFiltersToQueryParams } from 'core/searchUtils';
+import ToggleOnMobileCard from 'ui/components/ToggleOnMobileCard';
+
+import './styles.scss';
+
+
+export class SearchFiltersBase extends React.Component {
+  static propTypes = {
+    clientApp: PropTypes.string.isRequired,
+    filters: PropTypes.object.isRequired,
+    i18n: PropTypes.object.isRequired,
+    lang: PropTypes.string.isRequired,
+    pathname: PropTypes.string.isRequired,
+  }
+
+  static contextTypes = {
+    router: PropTypes.object,
+  }
+
+  onSelectElementChange = (event) => {
+    const { clientApp, filters, lang, pathname } = this.props;
+    event.preventDefault();
+
+    const newFilters = { ...filters };
+
+    // TODO: Make this better; paginator resets it anyway
+    delete newFilters.clientApp;
+
+    // Get the filter we're supposed to change and set it.
+    const filterName = event.currentTarget.getAttribute('data-filterName');
+    newFilters[filterName] = event.currentTarget.value;
+
+    // If the filters haven't changed we're not going to change the URL.
+    if (newFilters[filterName] === filters[filterName]) {
+      log.debug(oneLine`onSelectElementChange() called in SearchFilters but
+        the filter ${filterName} did not change–not changing route.`);
+      return false;
+    }
+
+    // If we got a `null` value it means we should clear this filter.
+    // This usually means "All".
+    if (newFilters[filterName] === '') {
+      delete newFilters[filterName];
+    }
+
+    // TODO: We do this in a few places; make a helper for it.
+    this.context.router.push({
+      pathname: `/${lang}/${clientApp}${pathname}`,
+      query: convertFiltersToQueryParams(newFilters),
+    });
+
+    return false;
+  }
+
+  addonTypeOptions() {
+    const { i18n } = this.props;
+
+    return [
+      // HACK: Should we be using at least `null` here or will that
+      // be oddly cast?
+      { children: i18n.gettext('All'), value: '' },
+      { children: i18n.gettext('Extension'), value: ADDON_TYPE_EXTENSION },
+      { children: i18n.gettext('Theme'), value: ADDON_TYPE_THEME },
+    ];
+  }
+
+  operatingSystemOptions() {
+    const { i18n } = this.props;
+
+    return [
+      // HACK: Should we be using at least `null` here or will that
+      // be oddly cast?
```

Comment:
I think an empty string value makes sense since it's mapping to query string params.